### ISSUE TITLE
Enabling the `threads` option makes it so that the plugin doesn't emit errors unless in watch mode after the file was changed.

### ISSUE THREAD

**Issue by HristoKolev** [webpack-contrib/eslint-webpack-plugin#146]
### Bug report

Given these options: 

```json
{
      extensions: ['js', 'jsx', 'ts', 'tsx'],
      failOnWarning: true,
      threads: true,
}
```

when I run webpack the plugin doesn't emit errors.

When I run webpack in watch mode the plugin doesn't emit errors initially but if I change the file containing the error the plugin emits the error.

If I change the `threads` option to `false` it works as expected and emits errors when webpack is run.

### Please paste the results of `npx webpack-cli info` here, and mention other relevant information

```

  System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
    Memory: 16.87 GB / 31.94 GB
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - C:\Program Files\nodejs\yarn.CMD
    npm: 8.5.5 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 99.0.4844.82
    Edge: Spartan (44.19041.1266.0), Chromium (99.0.1150.55)
  Packages:
    babel-loader: ^8 => 8.2.4
    browser-sync-webpack-plugin: ^2 => 2.3.0
    clean-webpack-plugin: ^4 => 4.0.0
    css-loader: ^6 => 6.7.1
    eslint-webpack-plugin: ^3 => 3.1.1
    fork-ts-checker-webpack-plugin: ^7 => 7.2.1
    html-loader: ^3 => 3.1.0
    html-webpack-plugin: ^5 => 5.5.0
    postcss-loader: ^6 => 6.2.1
    resolve-url-loader: ^5 => 5.0.0
    sass-loader: ^12 => 12.6.0
    style-loader: ^3 => 3.3.1
    ts-loader: ^9 => 9.2.8
    tsconfig-paths-webpack-plugin: ^3 => 3.5.2
    url-loader: ^4 => 4.1.1
    webpack: ^5 => 5.70.0
    webpack-bundle-analyzer: ^4 => 4.5.0
    webpack-cli: ^4 => 4.9.2

```

**Comment by JUST-Limbo** [comment]
Encountered  same problem the other day,it looks like a bug.
It was mentioned by other developer's comment in  #137.

**Comment by kirill-martynov** [comment]
I've got same problem, when threads: true, there is no errors and warnings

**Comment by xiayongtian** [comment]
对的，一样，当线程：true，没有错误和警告

**Comment by ajiho** [comment]
Until 2023, this issue still exists

**Comment by avs-doom** [comment]
Yes, still relevant and works similar to the "lintDirtyModulesOnly" setting
